### PR TITLE
Update Search to use Garden's backend (instead of Globus Search endpoint)

### LINKED ISSUE #161: Productionized full-text search in Postgres
Hayden showed that we can plausibly replace the search index with Postgres in https://github.com/Garden-AI/garden-backend/issues/154. This ticket will track the work of taking the new search endpoint from prototype to prod deployment.

Things to consider:
- I'd like to be confident that this solution can support on the order of 1k gardens.
- Do we need to add any indexing upfront?

### LINKED ISSUE #205: Point the frontend to the new backend search route
Sister ticket to https://github.com/Garden-AI/garden-backend/issues/161

We have a good prototype of this across BE and FE. This ticket tracks the work of any further tweaks needed to productionize and ship it.

### PR DESCRIPTION
**Pull request by shane-jung**
## Overview

This PR points the frontend to the backend's new search route, replacing Globus Search. No significant changes were made to the functionality or the appearance of the search page. 

### Screenshots
image

### CODE REVIEW

**Comment by shane-jung** [comment]
This should be ready, but should be released in unison with the backend search updates.

Edit: Still waiting on backend to be merged (to pull down changes to the backend schema, currently we're just using fields that were manually typed on the front end)

File: garden/src/hooks/useSearchResults.ts [hunk 1]
```diff
@@ -120,13 +126,29 @@ export const useSearchResults = (): SearchResultsState => {
...
+  const facets = useMemo(() => {
+    if (!searchResult?.facets) return [];
+    return Object.entries(searchResult.facets).map(
+      ([name, values]: [string, Record<string, number>]) => ({
+        name,
+        values: Object.entries(values)
+          .map(([value, count]: [string, number]) => ({ value, count }))
+          .sort((a, b) => {
+            return (
+              2 * (b.count - a.count) - (Number(selectedFilters[name]?.includes(a.value)) || -1)
```
Comment (WillEngler) [inline garden/src/hooks/useSearchResults.ts]: This is a little hard to follow. I don't understand why we're multiplying by 2 and being inside this nested map and sort makes it harder. Could you break this up and/or comment it a bit to convey what's going on?

File: garden/src/api/gardens/useSearchGardens.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
+import { search } from "@globus/sdk/cjs";
```
Comment (WillEngler) [inline garden/src/api/gardens/useSearchGardens.ts]: Are we actually using the Globus SDK for the search logic here?

**Review by WillEngler — COMMENTED** [review]
Looking solid overall, just leaving some requests for clarification before approving

File: garden/src/hooks/useSearchResults.ts (hunk 1 above)
Comment (shane-jung) [reply garden/src/hooks/useSearchResults.ts]: Yes this was indeed a bit strange. Basically this is to ensure that _selected_ filters appear at the top of the filters list. But I also want filters with a high number of matching results to appear near the top. I'll  try to write this in a way that's more clear

File: garden/src/api/gardens/useSearchGardens.ts (hunk 2 above)
Comment (shane-jung) [reply garden/src/api/gardens/useSearchGardens.ts]: Looks like this was left over from the previous version, I'll remove it in the next commit

**Review by WillEngler — APPROVED** [review]
LGTM! Thanks, Shane
